dgp/lib/plots.py

```diff
@@ -93,38 +92,85 @@ def mapplot_line(pnt_full, pnt, data, var, units='', ptitle='test_map', pfile='t
+    ax = plt.axes(projection=myproj)
+
+    s1 = plt.scatter(pnt_full['long'], pnt_full['lat'], c='black', s=1, transform=ccrs.PlateCarree())
+    s2 = plt.scatter(pnt['long'], pnt['lat'], c=data, cmap=cm.Spectral, s=10, transform=ccrs.PlateCarree())
+    p1 = ax.plot(pnt['long'][0], pnt['lat'][0], 'k*', markersize=7, transform=ccrs.PlateCarree())
+    cb = fig.colorbar(s2, ax=ax, label=units,
+                      orientation='vertical', shrink=0.8, pad=0.05)
+    cb.ax.set_yticklabels(cb.ax.get_yticklabels(), rotation=0)
+    cb.ax.tick_params(labelsize=6)
+
+    ax.coastlines(resolution='10m')
+    ax.xlabels_top = ax.ylabels_right = False
+    ax.gridlines(draw_labels=False, alpha=0.3, color='grey')
+    ax.xformatter = LONGITUDE_FORMATTER
+    ax.yformatter = LATITUDE_FORMATTER
+    ax.set_extent([x0, x1, y0, y1], ccrs.PlateCarree())
+
+    plt.tight_layout()
+    plt.subplots_adjust(top=0.90)
+    plt.suptitle(ptitle, y=0.98)
+    plt.savefig(pfile, bbox_inches='tight')  # save the figure to file
+    plt.close()
+
     return
+
+
+def mapplot_segment(df, var, begin_segment, end_segment, units='', ptitle='test_map', pfile='test_map'):
+    """
+    This makes a map plot of line segment
+    :param df:
+    :param var:
+    :param units:
+    :param ptitle:
+    :param pfile:
+    :return:
+    """
+
+    data = df[var][(begin_segment <= df[var].index) & (df[var].index <= end_segment)]
+    lat = df['trajectory'].lat[(begin_segment <= df['trajectory'].lat.index)
+                               & (df['trajectory'].lat.index <= end_segment)]
+    long = df['trajectory'].long[(begin_segment <= df['trajectory'].long.index)
+                                 & (df['trajectory'].long.index <= end_segment)]
+    box = sgeom.box(minx=long.min() - 1, maxx=long.max() + 1,
+                    miny=lat.min() - 1, maxy=lat.max() + 1)
+    x0, y0, x1, y1 = box.bounds
+    if x0 < 0:
+        myproj = ccrs.SouthPolarStereo(central_longitude=180)
+    else:
+        myproj = ccrs.NorthPolarStereo(central_longitude=0)
+    fig = plt.figure(figsize=(8, 4), facecolor='white', dpi=144)
+    ax = plt.axes(projection=myproj)
+
+    s2 = plt.scatter(long, lat, c=data, cmap=cm.Spectral, s=10, transform=ccrs.PlateCarree())
+    cb = fig.colorbar(s2, ax=ax, label=units,
+                      orientation='vertical', shrink=0.8, pad=0.05)
+    cb.ax.set_yticklabels(cb.ax.get_yticklabels(), rotation=0)
+    cb.ax.tick_params(labelsize=6)
+
+    ax.coastlines(resolution='10m')
+    ax.xlabels_top = ax.ylabels_right = False
+    ax.gridlines(draw_labels=False, alpha=0.3, color='grey')
+    ax.xformatter = LONGITUDE_FORMATTER
+    ax.yformatter = LATITUDE_FORMATTER
+    # ax.set_extent([x0, x1, y0, y1], ccrs.PlateCarree())
+
+    plt.tight_layout()
+    plt.subplots_adjust(top=0.90)
+    plt.suptitle(ptitle, y=0.98)
+    plt.savefig(pfile, bbox_inches='tight')  # save the figure to file
+    plt.close()
+
+    return
```

Comment:
Nothing to hold this up, but for future reference, there is an implicit `return None` at the end of a function if it's not put there explicitly. So no need for this.